### PR TITLE
COR-466: recipient form validation

### PR DESCRIPTION
**Pull request by internetti**
[COR-466]

validate recipient forms on submit

[COR-466]: https://dteam-nrc.atlassian.net/browse/COR-466?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: frontend/apps/pwa/src/features/former/validation.ts [hunk 1]
```diff
@@ -131,6 +131,24 @@ export const customValidation = {
...
+      if (keyFields.length !== 1) {
+        errors.push({
+          field: 'fields' as FieldPath<ValidationForm>,
+          message: 'Form needs to have exactly 1 key field',
+        });
+      } else if (keyFields[0].fieldType !== FieldKind.Reference) {
+        console.log('key fields', keyFields);
+
+        errors.push({
+          field: 'fields' as FieldPath<ValidationForm>,
```
Comment (neb42) [inline frontend/apps/pwa/src/features/former/validation.ts:147]: Not important, but I think `as const` should also work here. It'll turn the type from `string` to `fields`.

Comment (internetti) [reply frontend/apps/pwa/src/features/former/validation.ts:147]: seems to work, what benefit would the const version have? just shorter? less imports?

Comment (neb42) [reply frontend/apps/pwa/src/features/former/validation.ts]: Yeah, I'm not sure if there's a good reason to use one over the other. I personally use `const`, which as you say uses less imports. But I think both are fine.

**Pull request merged**